graphql/query_resolver.go

```diff
@@ -1026,17 +1034,19 @@ func (r *queryResolver) Waterfall(ctx context.Context, options WaterfallOptions)
 		activeVersionIds = append(activeVersionIds, v.Id)
 	}
 
-	waterfallVersions := groupInactiveVersions(activeVersionIds, allVersions)
+	waterfallVersions := groupInactiveVersions(allVersions)
+	bv := []*model.WaterfallBuildVariant{}
 
-	buildVariants, err := model.GetWaterfallBuildVariants(ctx, activeVersionIds)
-	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("getting waterfall build variants: %s", err.Error()))
-	}
+	if len(activeVersionIds) > 0 {
+		buildVariants, err := model.GetWaterfallBuildVariants(ctx, activeVersionIds)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("getting waterfall build variants: %s", err.Error()))
+		}
 
-	bv := []*model.WaterfallBuildVariant{}
-	for _, b := range buildVariants {
-		bCopy := b
-		bv = append(bv, &bCopy)
+		for _, b := range buildVariants {
+			bCopy := b
+			bv = append(bv, &bCopy)
+		}
 	}
 
 	// Return the min and max orders returned to be used as parameters for navigating to the next page
```

Comment:
I can't comment on the line since it's unchanged but should L1043 check if `allVersions` is empty? Or would there always be at least one version on the waterfall, even for a brand new project?

Reply:
Added!